meta.go

```diff
@@ -246,6 +267,63 @@ func (s *Store) key(cid *cid.Cid) datastore.Key {
 	return datastore.NewKey(cid.String())
 }
 
+// NewSwarmStore returns a new Swarm Store which uses an underlying datastore.
+func NewSwarmStore(serverURL string) (*SwarmStore, error) {
+	store, err := swarmdatastore.NewDatastore(serverURL)
```

Comment:
Why do we need both `SwarmStore` and `swarmdatastore.Datastore`? Can't we use the the Swarm datastore with the existing `Store` type?

Reply:
Why do we need both SwarmStore and swarmdatastore.Datastore?
 I though to keep it similar as possible to the ipfs datastore interface ..though we might not need that and can omit swarmdatastore.Datastore and be only with SwarmStore.
SwarmStore provides a bit different interface from ipfs Store . what will be the nicer way to merge it to one datastore interface ?

Reply:
I'm a little confused as to what you mean.

So we currently have the "META store" which has `Get(cid)` and `Put(obj)` and that wraps an underlying datastore interface which has `Get(key)` and `Put(key, value)`.

You have introduced both a "Swarm store" at the same level as the "META store" and also a "Swarm store" at the level of the underlying interface, and I am unsure why we need both?

I was expecting there to just be a new "Swarm store" as an underlying interface which could be used with the existing "META store", is that not possible?